geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java

```diff
@@ -63,6 +65,31 @@ private static ClientProtocolMessageHandler findClientProtocolMessageHandler() {
     }
   }
 
+  private static Class<? extends StreamAuthenticator> findStreamAuthenticator(
+      String implementationID) {
+    if (authenticatorClass != null) {
+      return authenticatorClass;
+    }
+
+    synchronized (streamAuthenticatorLoadLock) {
+      if (authenticatorClass != null) {
+        return authenticatorClass;
+      }
+
+      ServiceLoader<StreamAuthenticator> loader = ServiceLoader.load(StreamAuthenticator.class);
+
+      for (StreamAuthenticator classInstance : loader) {
+        if (implementationID.equals(classInstance.implementationID())) {
+          return classInstance.getClass();
```

Comment:
Why do we get an instance of the StreamAuthenticator, just to call `getClass` on it, and then later we create a new instance of class from this instance.